src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/data/ExternalMovementRepositoryTest.kt

```diff
@@ -0,0 +1,294 @@
+import org.springframework.test.context.ActiveProfiles
+import uk.gov.justice.digital.hmpps.digitalprisonreportingmi.data.ExternalMovementRepositoryTest.AllMovements.allExternalMovements
+import uk.gov.justice.digital.hmpps.digitalprisonreportingmi.model.ExternalMovementFilter
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.*
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test")
+class ExternalMovementRepositoryTest {
+
+  @Autowired
+  lateinit var externalMovementRepository: ExternalMovementRepository
+
+  @BeforeEach
+  fun setup() {
+    allExternalMovements.forEach {
+      externalMovementRepository.save(it)
+    }
+  }
+
+  @Test
+  fun `should return 2 external movements for the selected page 2 and pageSize 2 sorted by date in ascending order`() {
+    val actual = externalMovementRepository.list(2, 2, "date", true, emptyMap())
+    Assertions.assertEquals(listOf(AllMovements.externalMovement3, AllMovements.externalMovement4), actual)
+    Assertions.assertEquals(2, actual.size)
+  }
+
+  @Test
+  fun `should return 1 external movement for the selected page 3 and pageSize 2 sorted by date in ascending order`() {
+    val actual = externalMovementRepository.list(3, 2, "date", true, emptyMap())
+    Assertions.assertEquals(listOf(AllMovements.externalMovement5), actual)
+    Assertions.assertEquals(1, actual.size)
+  }
+
+  @Test
+  fun `should return 5 external movements for the selected page 1 and pageSize 5 sorted by date in ascending order`() {
+    val actual = externalMovementRepository.list(1, 5, "date", true, emptyMap())
+    Assertions.assertEquals(listOf(AllMovements.externalMovement1, AllMovements.externalMovement2, AllMovements.externalMovement3, AllMovements.externalMovement4, AllMovements.externalMovement5), actual)
+    Assertions.assertEquals(5, actual.size)
+  }
+
+  @Test
+  fun `should return an empty list for the selected page 2 and pageSize 5 sorted by date in ascending order`() {
+    val actual = externalMovementRepository.list(2, 5, "date", true, emptyMap())
+    Assertions.assertEquals(emptyList<ExternalMovementEntity>(), actual)
+  }
+
+  @Test
+  fun `should return an empty list for the selected page 6 and pageSize 1 sorted by date in ascending order`() {
+    val actual = externalMovementRepository.list(6, 1, "date", true, emptyMap())
+    Assertions.assertEquals(emptyList<ExternalMovementEntity>(), actual)
+  }
+
+  @TestFactory
+  fun `should return all external movements for the selected page and pageSize sorted by date when sortedAsc is true and when it is false`() =
+    assertExternalMovements(sortColumn = "date", expectedForAscending = AllMovements.externalMovement1, expectedForDescending = AllMovements.externalMovement5)
+
+  @TestFactory
+  fun `should return all external movements for the selected page and pageSize sorted by time when sortedAsc is true and when it is false`() =
```

Comment:
Note here this tests that sorting is based only on the time part regardless of the date.